packages/common/src/projects/types.ts

```diff
@@ -0,0 +1,146 @@
+import { IUserRequestOptions } from "@esri/arcgis-rest-auth";
+import { IUserItemOptions } from "@esri/arcgis-rest-portal";
+import { IRequestOptions } from "@esri/arcgis-rest-request";
+import { IGeometry, IPoint } from "@esri/arcgis-rest-types";
+
+export type IHubProjectOptions = Pick<Partial<IHubProject>, "title" | "org">;
+
+/**
+ * Base properties for Hub Models
+ * This is a subset of IItem, that can apply to
+ * models that are not backed by items
+ */
+export interface IHubBaseModel {
+  readonly id?: string;
+  readonly owner?: string;
+  title: string;
+  description?: string;
+  snippet?: string;
+  extent?: number[][];
+  culture?: string;
+  properties?: any;
+  url?: string;
+  tags?: string[];
+  typeKeywords?: string[];
+  readonly created?: number;
+  readonly modified?: number;
+}
+
+/**
+ * Defines the properties of a Hub Project object
+ */
+export interface IHubProject extends IHubBaseModel, IWithSlug, IWithLayout {
+  timeline?: IHubTimeline;
+}
+
+export interface IWithSlug {
+  slug?: string;
+  org: { id: string; key: string };
+}
+
+export interface IWithLayout {
+  layout?: IHubLayout;
+}
+
+/**
+ * Defined layout related functions that a module
+ * working with IWithLayout objects must implement
+ */
+export interface IWithLayoutFns {
+  listDrafts(id: string, ro: IRequestOptions): Promise<string[]>;
+  destroyDraft(
+    id: string,
+    name: string,
+    ro: IUserRequestOptions
+  ): Promise<void>;
+  saveDraft(
+    id: string,
+    name: string,
+    layout: IHubLayout,
+    ro: IUserRequestOptions
+  ): Promise<void>;
+  loadDraft(id: string, name: string, ro: IRequestOptions): Promise<IHubLayout>;
+}
+
+// Stored in item.data.heroImage
+// can be resource or url
+// Module needs to implement
+export interface IWithHeroImage {
```

Comment:
minor - but "Hero Image" is confusing vernacular that I don't think is common. Can we use "Banner Image" instead?

Reply:
We can discuss. Items have `banner` but the docs suggest it's a url to an image to be used for the banner. Very unclear if this is something we upload or link to. I'm cool w/ calling it banner, but it _seems_ like we need to store it as a resource.